compiler/test/compilable/test24479.d

```diff
@@ -0,0 +1,25 @@
+// https://issues.dlang.org/show_bug.cgi?id=24479
+
+struct S
+{
+    S opBinary(string op: "-")(S rhs) const pure nothrow @nogc
+    {
+        return rhs;
+    }
+    S opBinary(string op: "*")(S dur) const pure nothrow @nogc
+    {
+        return dur;
+    }
+}
+
+private enum hasExternalUDA(alias A) = is(A == External) || is(typeof(A) == External);
+
+void foo()
+{
+    static foreach (t; __traits(getOverloads, S, "opBinary", true))
+        static foreach(attr; __traits(getAttributes, t))
+            pragma(msg, attr);
+
+    static assert(__traits(getOverloads, S, "opBinary", true).length == 2);
+    alias A = __traits(getAttributes, __traits(getOverloads, S, "opBinary", true)[1]);
```

Comment:
in addition to testing that the call can be made, I would also add tests to verify that it indeed retrieves attributes of those overloads.  Loosely related to https://issues.dlang.org/show_bug.cgi?id=24480